frontend/src/store/teachers.js

```diff
@@ -87,4 +87,49 @@ export const teacherGetters = {
   teachers(state) {
     return state.teachers;
   },
+
+  suggestedTeachers(state) {
+    return (studentId) => {
+
+      // 1. Find the student using the ID
+      const student = state.students.find(student => student.StudentID === studentId)
+      if (!student) {
+        return []
+      }
+      // 2. Find their native language
+      const studentNativeLanguage = student.NativeLanguageID
+      
+      // 3. Get unmatched teachers whose first or second language equals student's native language
+      const relevantTeachers = state.teachers.filter(teacher => {
+        const doesNativeLanguageMatch = teacher.NativeLanguageID === studentNativeLanguage
+        const doesSecondLanguageMatch = teacher.SecondLanguageID === studentNativeLanguage
+
+        return ((doesNativeLanguageMatch || doesSecondLanguageMatch))
```

Comment:
should `SecondLanguageProficiency` factor in here?

Reply:
Good question... it's not referred to in the requirements, so for now as long as there is a match on second language, that teacher is a candidate. I'm inclined to leave it as that for now and get these changes merged in. Refinements to the suggested teachers "algo" can be iterated on.